### PR TITLE
Implement PatchingDeletedDocumentsDoesNothing test

### PR DESCRIPTION
**Pull request by rsgowman**
...and add placeholders for numeric transform tests

### CODE REVIEW

File: Firestore/core/test/firebase/firestore/model/mutation_test.cc [hunk 1]
```diff
@@ -148,13 +148,43 @@ TEST(Mutation, PatchesPrimitiveValue) {
 }
 
 TEST(Mutation, PatchingDeletedDocumentsDoesNothing) {
-  // TODO(rsgowman)
+  auto base_doc =
+      std::make_shared<NoDocument>(testutil::DeletedDoc("collection/key", 0));
```
Comment (var-const) [inline Firestore/core/test/firebase/firestore/model/mutation_test.cc]: Question: are documents always managed by a `shared_ptr`? If yes, perhaps `testutil::DeletedDoc` should return a shared pointer?

Comment (wilhuff) [reply Firestore/core/test/firebase/firestore/model/mutation_test.cc]: Seconded. Our main goal with `testutil` helpers is keep noise out of the tests. Anything we can do to keep the test specification concise helps keep them comprehensible.

Comment (rsgowman) [reply Firestore/core/test/firebase/firestore/model/mutation_test.cc]: > Question: are documents always managed by a shared_ptr?

No, but they usually are. On the balance, this is probably worth it. It's orthogonal to this change so I'll submit this, and then create a followup.

Comment (rsgowman) [reply Firestore/core/test/firebase/firestore/model/mutation_test.cc]: #2546

**Pull request merged**